### PR TITLE
Add `CGColorSpace` type

### PR DESCRIPTION
**Pull request by jinleili**
Prepare to add color space to wgpu

### CODE REVIEW

File: src/color_space.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    // CG_AVAILABLE_STARTING(10.14.3, 12.3)
+    static kCGColorSpaceExtendedLinearDisplayP3: CFStringRef;
+
+    // HDR
+    // CG_AVAILABLE_STARTING(12.0, 15.0)
+    static kCGColorSpaceLinearITUR_2020: CFStringRef;
+    // CG_AVAILABLE_BUT_DEPRECATED(10.15.4, 11.0, 13.4, 14.0);
+    static kCGColorSpaceITUR_2020_PQ: CFStringRef;
+    // CG_AVAILABLE_BUT_DEPRECATED(10.15.6, 11.0, 12.6, 14.0)
+    static kCGColorSpaceITUR_2020_HLG: CFStringRef;
```
Comment (cwfitzgerald) [inline src/color_space.rs:75]: Let's get rid of these, There's no difference between these and the 2100 ones, the 2020 ones were just mildly incorrectly named.

File: src/color_space.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    // CG_AVAILABLE_STARTING(10.12, 10.0)
+    static kCGColorSpaceExtendedLinearSRGB: CFStringRef;
+    // CG_AVAILABLE_STARTING(11.0, 14.0)
+    static kCGColorSpaceExtendedDisplayP3: CFStringRef;
+    // CG_AVAILABLE_STARTING(10.14.3, 12.3)
+    static kCGColorSpaceExtendedLinearDisplayP3: CFStringRef;
+
+    // HDR
+    // CG_AVAILABLE_STARTING(12.0, 15.0)
+    static kCGColorSpaceLinearITUR_2020: CFStringRef;
```
Comment (cwfitzgerald) [inline src/color_space.rs:71]: We seem to be missing https://developer.apple.com/documentation/coregraphics/cgcolorspace/3892090-itur_2020_srgbgamma, which does come up

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Looking good, some comments.

**Review by madsmtm — COMMENTED** [review]
Just noting: Somewhat duplicate of what's in https://github.com/servo/core-foundation-rs/blob/de8752262a2f4a253c8c0f214294094c28e9c0ba/core-graphics/src/color_space.rs

**Comment by jinleili** [comment]
Yes, but metal-rs currently does not directly depend on `core-graphics` crate, just `core-graphics-types`

**Comment by AlexanderEkdahl** [comment]
@jinleili I added these color spaces in https://github.com/servo/core-foundation-rs/pull/533 which would allow your implementation to use extended range rendering if you depend on `core-graphics`.

**Pull request closed without merging**